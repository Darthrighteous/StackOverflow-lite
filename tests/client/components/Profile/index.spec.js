import React from 'react';
import { shallow } from 'enzyme';
import configureMockStore from 'redux-mock-store';

import ConnectedProfile, { Profile } from '../../../../client/components/Profile';
import { mockQuestionsResponse } from '../../../support/mockData';

const signedInUser = {
  answer_count: 5,
  comment_count: 8,
  downvoted_answers: [1, 6, 7],
  downvoted_questions: [1, 2],
  email: "ja.ogunniyi@test.com",
  firstname: "Jalil",
  id: 22,
  joined_at: "2018-12-26T08:24:09.197Z",
  lastname: "Ogunniyi",
  question_count: 1,
  upvoted_answers: [1],
  upvoted_questions: [2, 1],
  username: "Darthrighteous"
};

describe('Profile component test', () => {
  afterEach(() => {
    localStorage.clear();
  });

  const mockStore = configureMockStore([]);
  const store = mockStore({
    global: {
      isLoggedIn: true
    },
    questions: [],
    userQuestions: [],
    user: { username: 'Darthrighteous' }
  });

  const props = {
    userQuestions: mockQuestionsResponse.questions,
    profileOwner: {},
    match: { params: { username: 'Darthrighteous' } },
    fetchSingleUser: jest.fn(),
    fetchUserQuestions: jest.fn(),
  };

  test('snapshot test', () => {
    const ProfileWrapper = shallow(<Profile {...props} />);
    expect(ProfileWrapper).toMatchSnapshot();

    const instance = ProfileWrapper.instance();
    instance.setSort('test sort');
    expect(instance.state.sort).toBe('test sort');
    instance.setSort('test sort');
    expect(instance.state.sort).toBe('test sort');
  });

  test('connected componenet snapshot test', () => {
    const connectedProfile = shallow(
      <ConnectedProfile {...props} store={store} />
    );
    expect(connectedProfile).toMatchSnapshot();
  });

  test('owner profile snapshot', () => {
    localStorage.setItem('user', JSON.stringify(signedInUser));
    const ownerProps = {
      ...props,
      userQuestions: [],
      profileOwner: {
        username: 'Darthrighteous',
      },
    };
    const ownerWrapper = shallow(<Profile {...ownerProps} />);
    expect(ownerWrapper).toMatchSnapshot();
  });

  test('with questions snapshot', () => {
    const withQuestionsProps = {
      ...props,
      userQuestions: [],
      profileOwner: { username: 'tracy'},
    };
    const withQuestions = shallow(<Profile {...withQuestionsProps} />);
    expect(withQuestions).toMatchSnapshot();
  });
});
