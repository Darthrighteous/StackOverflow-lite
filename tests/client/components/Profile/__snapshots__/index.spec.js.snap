// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component test connected componenet snapshot test 1`] = `
<Profile
  fetchSingleUser={[Function]}
  fetchUserQuestions={[Function]}
  match={
    Object {
      "params": Object {
        "username": "Darthrighteous",
      },
    }
  }
  profileOwner={
    Object {
      "username": "Darthrighteous",
    }
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  userQuestions={Array []}
/>
`;

exports[`Profile component test owner profile snapshot 1`] = `
<div
  className="main-content"
>
  <div
    className="profile-content"
  >
    <div
      className="profile-primary"
    >
      <ContentOptions
        background={false}
        onSortClick={[Function]}
        sort="most recent"
        title="Your Questions"
      />
      <div
        className="no-resource-message"
      >
        <span>
          No questions here yet
        </span>
      </div>
    </div>
    <div
      className="profile-side"
    >
      <div
        className="user-details"
      >
        <i
          className="fas fa-user-circle"
        />
        <div
          className="user-name-container"
        >
          <span
            id="fullname"
          >
            undefined undefined
          </span>
          <span
            id="username"
          >
            Darthrighteous
          </span>
        </div>
      </div>
      <div
        className="btn-new-question-cont"
      >
        <ButtonNewQuestion />
      </div>
      <div
        className="user-stats"
      >
        <div
          className="stats-asked"
        >
          <h5>
            Asked
          </h5>
          <span
            id="question-count"
          />
        </div>
        <div
          className="stats-answered"
        >
          <h5>
            Answered
          </h5>
          <span
            id="answer-count"
          />
        </div>
        <div
          className="stats-comments"
        >
          <h5>
            Comments
          </h5>
          <span
            id="comment-count"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Profile component test snapshot test 1`] = `
<div
  className="main-content"
>
  <div />
</div>
`;

exports[`Profile component test with questions snapshot 1`] = `
<div
  className="main-content"
>
  <div
    className="profile-content"
  >
    <div
      className="profile-primary"
    >
      <ContentOptions
        background={false}
        onSortClick={[Function]}
        sort="most recent"
        title="Questions by tracy "
      />
      <div
        className="no-resource-message"
      >
        <span>
          No questions here yet
        </span>
      </div>
    </div>
    <div
      className="profile-side"
    >
      <div
        className="user-details"
      >
        <i
          className="fas fa-user-circle"
        />
        <div
          className="user-name-container"
        >
          <span
            id="fullname"
          >
            undefined undefined
          </span>
          <span
            id="username"
          >
            tracy
          </span>
        </div>
      </div>
      <div
        className="btn-new-question-cont"
      >
        <ButtonNewQuestion />
      </div>
      <div
        className="user-stats"
      >
        <div
          className="stats-asked"
        >
          <h5>
            Asked
          </h5>
          <span
            id="question-count"
          />
        </div>
        <div
          className="stats-answered"
        >
          <h5>
            Answered
          </h5>
          <span
            id="answer-count"
          />
        </div>
        <div
          className="stats-comments"
        >
          <h5>
            Comments
          </h5>
          <span
            id="comment-count"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
